include/aws/sdkutils/private/endpoints_eval_util.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#ifndef AWS_SDKUTILS_ENDPOINTS_EVAL_UTIL_H
+#define AWS_SDKUTILS_ENDPOINTS_EVAL_UTIL_H
+
+#include <aws/sdkutils/sdkutils.h>
+
+struct aws_string;
+struct aws_byte_buf;
+struct aws_json_value;
+
+struct aws_byte_cursor aws_byte_cursor_from_substring(const struct aws_string *src, size_t start, size_t end);
+
+/*
+ * Replace escaped chars within endpoints templated strings.
+ * Basically replaces {{ with { and }} with }.
+ * Note: this function does not care about existence of { or } and will
+ * leave them as is.
+ */
+AWS_SDKUTILS_API int aws_templated_string_replace_escaped(
+    struct aws_allocator *allocator,
+    struct aws_byte_cursor str,
+    struct aws_byte_buf *out_buf);
```

Comment:
document that this will initialize `out_buf` if it succeeds, and if it fails then out_buf will remain uninitialized

or you could rename this like an aws_byte_buf_init() function so the lifetimes bit is more clear:

`int aws_byte_buf_init_from_templated_string_replace_escaped(struct aws_byte_buf *dest, struct aws_allocator*, struct aws_byte_cursor src)`